src/ProjectTemplates/BlazorTemplates.Tests/BlazorTemplateTest.cs

```diff
@@ -1,49 +1,146 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
 using System.Runtime.InteropServices;
+using System.Threading.Tasks;
 using Microsoft.AspNetCore.BrowserTesting;
+using Microsoft.AspNetCore.Testing;
+using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Logging.Testing;
-using ProjectTemplates.Tests.Infrastructure;
+using Templates.Test.Helpers;
 using Xunit;
 using Xunit.Abstractions;
 
 namespace Templates.Test
 {
-    public class BlazorTemplateTest
+    public abstract class BlazorTemplateTest : LoggedTest, IAsyncLifetime
     {
-        public BlazorTemplateTest(PlaywrightFixture<BlazorServerTemplateTest> browserFixture)
+        public const int BUILDCREATEPUBLISH_PRIORITY = -1000;
+
+        public BlazorTemplateTest(ProjectFactoryFixture projectFactory)
+        {
+            ProjectFactory = projectFactory;
+        }
+
+        public ProjectFactoryFixture ProjectFactory { get; set; }
+        public ContextInformation BrowserContextInfo { get; protected set; }
+        public BrowserManager BrowserManager { get; private set; }
+
+        private ITestOutputHelper _output;
+        public ITestOutputHelper Output
         {
-            Fixture = browserFixture;
+            get
+            {
+                if (_output == null)
+                {
+                    _output = new TestOutputLogger(Logger);
+                }
+                return _output;
+            }
         }
+        
+        public abstract string ProjectType { get; }
+        private static readonly bool _isCIEnvironment =
+            !string.IsNullOrWhiteSpace(Environment.GetEnvironmentVariable("ContinuousIntegrationBuild"));
+
+        protected async override Task InitializeCoreAsync(TestContext context)
+        {
+            BrowserManager = await BrowserManager.CreateAsync(CreateConfiguration(), LoggerFactory);
```

Comment:
As per discussion, you will undo this after we get this in? (revert to it being a single fixture)

Reply:
Yep I'll file an issue, its got a complication in that the new nice file logging per test only works well as a side effect of this split too, so its going to be not completely trivial

Reply:
I'm fine with that, as long as we push all the playwright logs into its own debug file, that's good enough for me.

Reply:
https://github.com/dotnet/aspnetcore/issues/31111